python/src/trezorlib/cli/firmware.py

```diff
@@ -593,11 +639,10 @@ def update(
     dry_run: bool,
     beta: bool,
     bitcoin_only: Optional[bool],
```

Comment:
the flow is super confusing and I thing the whole function should be rethought and refactored.

initial idea, high-level steps (probably worth making a separate function each)

1. figure out what firmware to install, return already validated firmware bytes
2. jump to bootloader -- if not already there, invoke ILU. Thinking deeper about this, I'm honestly not sure if the whole ILU detection makes any sense? We can _always_ invoke the ILU call if we're in firmware, and the older ones will simply not recognize the new options and drop you into BL as normal. So let's just do that.
3. upload the firmware that we got from step (1)

(1) and (2) can share the same client instance. (3) needs a new one, so that's two `with client_context` blocks

the wait loop between (2) and (3) can be written like this:

```
click.echo("Waiting for bootloader...")
while True:
    time.sleep(0.5)
    try:
        obj.get_transport()
        break
    except Exception:
        pass

with obj.client_context() as client:
    if not client.bootloader_mode:
        click.echo("Failed to enter bootloader mode, please try again.")
        sys.exit(1)
```